pkg/solana/logpoller/types.go

```diff
@@ -126,19 +127,14 @@ type Decoder interface {
 	Decode(_ context.Context, raw []byte, into any, itemType string) error
 }
 
-type EventIdl struct {
-	codec.EventIDLTypes
-}
+type EventIdl codec.EventIDLTypes
 
 func (e *EventIdl) Scan(src interface{}) error {
 	return scanJSON("EventIdl", e, src)
 }
 
 func (e EventIdl) Value() (driver.Value, error) {
-	return json.Marshal(map[string]any{
-		"IdlEvent":        e.EventIDLTypes.Event,
-		"IdlTypeDefSlice": e.EventIDLTypes.Types,
-	})
+	return json.Marshal(e)
```

Comment:
Is there a reason why we need to save the rest of the junk in the idl? AFAIK we should only need these two fields the rest just takes up unnecessary space. Ideally, ChainReader should remove the extra fields before it's passed to LogPoller, but I added this so at least we're not persisting them.

Reply:
Oh, is it because this was messing up the `(e EventIdl) Equal` method? If that's the only reason... we could change that to compare those two fields instead of the whole thing.

Reply:
`codec.EventIDLTypes` has only two fileds. Ilija changed it in another PR.

Reply:
Oh, nice!